sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/epkversion/PartitionProcessorImpl.java

```diff
@@ -248,7 +271,39 @@ public Mono<Void> run(CancellationToken cancellationToken) {
                                     }).flatMap(values -> Flux.empty());
                             }
                         }
-                        break;
+                        case PARSING_ERROR:
+                            if (this.unparseableDocumentRetries.incrementAndGet() == 1) {
+                                logger.warn(
+                                    "Lease with token {}: Attempting a retry on parsing error.",
+                                    this.lease.getLeaseToken());
+                                this.resultException = new RuntimeException(clientException);
+                                this.options.setMaxItemCount(1);
+                                return Flux.empty();
+                            } else {
+
+                                // No way to recover from this
+                                logger.error("Encountered parsing error which is not recoverable, attempting to skip document", clientException);
+
+                                String continuation = CosmosChangeFeedContinuationTokenUtils.extractContinuationTokenFromCosmosException(clientException);
```

Comment:
Wondering if we transfer etag or other response headers to the newly created cosmosException? or shall we preserve it while we have the response headers in the above flatMap?